pkgs/development/haskell-modules/hackage-packages.nix

```diff
@@ -158089,14 +158089,15 @@ self: {
 
   "libsodium" = callPackage
     ({ mkDerivation, base, hedgehog, libsodium, tasty, tasty-hedgehog
-     , tasty-hunit
+     , tasty-hunit, c2hs
```

Comment:
You can't edit this file since it is automatically generated.

Any overrides will have to go into `configuration-common.nix`.

Also, why is `c2hs` needed here?  Why doesn't this get picked up automatically?  Does the `libsodium.cabal` file not mention this dependency?

Reply:
Oh shoot! I did not realize this file was generated.

Let me analyze what's going on.

Reply:
Fixed it with a cabal override in `configuration-common.nix`

The reason we need to do this is because the author neglected to include c2hs in build-tool-depends in the package uploaded to hackage. It's fixed in their repo, but no release to hackage yet.

I've now written this in a style, that this only overrides libraryToolDepends if it's not specified already.